src/validator/CreditCardValidator.java

```diff
@@ -0,0 +1,62 @@
+package validator;
+
+import java.util.ArrayList;
+
+public class CreditCardValidator implements CreditCardChecking {
+
+	@Override
+	public boolean checkValidity(CreditCardVendor ccv, String number) {
+		ArrayList<Integer> chars = ccv.getNumberOfChars();
+		ArrayList<String> masks = ccv.getIIN();
+		
+		boolean result1 = checkLengthCorrectness(number, chars);
+		boolean result2 = checkIINMaskCorrectness (number, masks);
```

Comment:
It would be more elegant to name the results business-wise than result1 or result2. Easier to understand would be boolean lengthIsCorrect than result1. Similarly result2 and result3.